storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1446,6 +1448,22 @@ public static RuntimeException wrapInRuntime(Exception e){
         }
     }
 
+    public static int getAvailablePort(int prefferedPort) {
+        int localPort = -1;
+        try(ServerSocket socket = new ServerSocket(prefferedPort)) {
+            localPort = socket.getLocalPort();
+        } catch(IOException exp) {
+            if (prefferedPort > 0) {
+                return getAvailablePort(0);
+            }
+        }
+        return localPort;
```

Comment:
one minor comment. previous version never returns -1. can we just bubble up the exception in catch block if preferredPort is not greater than zero and not return a negative port? Other than that +1.